src/openms/include/OpenMS/DATASTRUCTURES/OSWData.h

```diff
@@ -0,0 +1,344 @@
+
+    */
+    class OPENMS_DLLAPI OSWPeptidePrecursor
+    {
+      public:
+        /// just a dummy feature to allow for acceptor output values etc
+        OSWPeptidePrecursor() = default;
+        /// custom c'tor which fills all the members with data; all members are read-only
+        OSWPeptidePrecursor(const String& seq, const short charge, const bool decoy, const float precursor_mz, std::vector<OSWPeakGroup>&& features);
+        /// Copy c'tor
+        OSWPeptidePrecursor(const OSWPeptidePrecursor& rhs) = default;
+        /// assignment operator
+        OSWPeptidePrecursor& operator=(const OSWPeptidePrecursor& rhs) = default;
+        /// move c'tor
+        OSWPeptidePrecursor(OSWPeptidePrecursor&& rhs) = default;
+        /// move assignment operator
+        OSWPeptidePrecursor& operator=(OSWPeptidePrecursor&& rhs) = default;
+
+        /// the peptide sequence (incl. mods)
+        const String& getSequence() const
+        {
+          return seq_;
+        }
+        /// precursor charge
+        short getCharge() const
+        {
+          return charge_;
+        }
+        /// is this a decoy feature (from a decoy protein)
+        bool isDecoy() const
+        {
+          return decoy_;
+        }
+        /// m/z of this charged peptide
+        float getPCMz() const
+        {
+          return precursor_mz_;
+        }
+        /// candidate explanations
+        const std::vector<OSWPeakGroup>& getFeatures() const
+        {
+          return features_;
+        }
+
+      private:
+        String seq_;
+        short charge_{0};
+        bool decoy_{false};
+        float precursor_mz_{0};
+        std::vector<OSWPeakGroup> features_;
+    };
+
+    /**
+      @brief A Protein is the highest entity and contains one or more peptides which were found/traced.
+
+    */
+    class OPENMS_DLLAPI OSWProtein
+    {
+      public:
+        /// just a dummy feature to allow for acceptor output values etc
+        OSWProtein() = default;
+        /// custom c'tor which fills all the members with data; all members are read-only
+        OSWProtein(const String& accession, const Size id, std::vector<OSWPeptidePrecursor>&& peptides);
+        /// Copy c'tor
+        OSWProtein(const OSWProtein& rhs) = default;
+        /// assignment operator
+        OSWProtein& operator=(const OSWProtein& rhs) = default;
+        /// move c'tor
+        OSWProtein(OSWProtein&& rhs) = default;
+        /// move assignment operator
+        OSWProtein& operator=(OSWProtein&& rhs) = default;
+
+        const String& getAccession() const
+        {
+          return accession_;
+        }
+
+        Size getID() const
+        {
+          return id_;
+        }
+
+        const std::vector<OSWPeptidePrecursor>& getPeptidePrecursors() const
+        {
+          return peptides_;
+        }
+
+      private:
+        String accession_;
```

Comment:
alignment?

Reply:
largest requirement is 8 and vector and string are both multiples of that.
So no matter where I put Size, I always loose 4 byte.